frame/support/procedural/src/pallet/parse/call.rs

```diff
@@ -228,17 +246,23 @@ impl CallDef {
 					weight_attrs.push(FunctionAttr::Weight(empty_weight));
 				}
 
-				if weight_attrs.len() != 1 {
-					let msg = if weight_attrs.is_empty() {
-						"Invalid pallet::call, requires weight attribute i.e. `#[pallet::weight($expr)]`"
-					} else {
-						"Invalid pallet::call, too many weight attributes given"
-					};
-					return Err(syn::Error::new(method.sig.span(), msg))
-				}
-				let weight = match weight_attrs.pop().unwrap() {
-					FunctionAttr::Weight(w) => w,
-					_ => unreachable!("checked during creation of the let binding"),
+				let weight = match weight_attrs.len() {
+					0 if inherited_call_weight.is_some() => CallWeightDef::Inherited,
+					0 if dev_mode => CallWeightDef::DevModeDefault,
```

Comment:
The question here is whether to prioritize `dev_mode` or the `call(weight = …)` when *both* are present.  
I now opted to use the specified weight instead of `dev_mode`.  

All the match statements in the expand code are exclusive, so no prioritization is needed.